app/models/service_template.rb

```diff
@@ -69,6 +69,70 @@ class ServiceTemplate < ApplicationRecord
   scope :with_existent_service_template_catalog_id, ->         { where.not(:service_template_catalog_id => nil) }
   scope :displayed,                                 ->         { where(:display => true) }
 
+  def self.catalog_item_types
+    CATALOG_ITEM_TYPES.each.with_object({}) do |(key, description), hash|
+      method_name = "#{key}_provider_present?".to_sym
+      hash[key] = { :description => description,
+                    :display     => ServiceTemplate.respond_to?(method_name) ? ServiceTemplate.send(method_name) : false }
+    end
+  end
+
+  def self.amazon_provider_present?
+    ExtManagementSystem.where(:type => "ManageIQ::Providers::Amazon::CloudManager").any?
+  end
+
+  def self.azure_provider_present?
+    ExtManagementSystem.where(:type => "ManageIQ::Providers::Azure::CloudManager").any?
+  end
+
+  def self.generic_provider_present?
+    true
+  end
+
+  ORCHESTRATION_PROVIDER_TYPES = %w(
+    ManageIQ::Providers::Amazon::CloudManager
+    ManageIQ::Providers::Azure::CloudManager
+    ManageIQ::Providers::Google::CloudManager
```

Comment:
no google, but we have `ManageIQ::Providers::VMWare::CloudManager`

Reply:
@bzwei Is there a way to dynamically build this list?

Reply:
@gmcculloug Yes we can. Working with Madhu now.